### DISCUSSION TITLE
Help trying to create a source.

### DISCUSSION THREAD

**Discussion by steveuk23** [mampfes/hacs_waste_collection_schedule#1508] [Q&A]
Hi guys.
Great integration thanks.
I thought I would have a go at trying to create a source for my area.
I'm not very good at python so have been guessing most of it but have some questions.
My area is <https://www.highpeak.gov.uk/findyourbinday> Mine requires a post code and then select a address so I've been using the derby source as a base and that uses "premises_id" and some others use "uprn" on my URL it shows "pageSessionId" when I enter in a postcode, is that what I should use ?
How do I know which one to use.
I've forked this to my GitHub and added the files but I actually have no idea how to run a test with the script.
Thanks for any help

**Comment by 5ila5** [comment]
Here is the contribution guide for a dedicated source https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_source.md

To test your source you add a TEST_CASES dictonary in your file (file should be named highpeak_gov_uk.py) then you can run `python custom_components/waste_collection_schedule/waste_collection_schedule/test/test_sources.py -s highpeak_gov_uk` I like to proved Argument -lit for debugging purposes (see all returned collection(l) including icons (i) show full error stacktrace (t))  

It looks like you need to use the postcode and uprn arguments and make post requests to https://www.highpeak.gov.uk/apiserver/formsservice/http/processsubmission and handling this weird sessionId and these other IDs.

looks like it's not the easiest to implement (like a lot of uk sources) you need some reverse engineering of the web requests (browser developer tools)

**Comment by 5ila5** [comment]
You max be able to use this implematation as a base https://github.com/robbrad/UKBinCollectionData/blob/master/uk_bin_collection/uk_bin_collection/councils/HighPeakCouncil.py
But you cannot use the selenium.webdriver part as this does not work well with HA and needs different workarounds for different installations

**Comment by steveuk23** [comment]
Thanks for the replies that's great I'll look into them.
